File: 프로그래머스/lv3/42579. 베스트앨범/베스트앨범.js
function solution(genres, plays) {
    const answer = [];
    const 앨범 = [];
    //hashtable
    for (let i = 0; i < genres.length; i++) {
        앨범.push({index: i, genre: genres[i], play: plays[i] });
    }
    // 재생순 합 구하기
    let 재생순장르={};
    for (let i = 0; i < genres.length; i++) {
        재생순장르[genres[i]] = 재생순장르[genres[i]] ? 재생순장르[genres[i]]+plays[i] : plays[i];
    }
    //재생순 정렬해서 최우선 장르 구하기
    const sort재생순=Object.entries(재생순장르)
    sort재생순.sort((a,b)=>b[1]-a[1]);

    sort재생순.map((장르) => {
        let arr = [];
        for (let i = 0; i < 앨범.length; i++) {
            if(장르[0] === 앨범[i].genre) arr.push(앨범[i])
        }
        //재생별로 정렬
        arr.sort((a,b) => b.play - a.play);
        const len = arr.length > 2 ? 2 : arr.length;
        for (let i = 0; i < len; i++) {
            answer.push(arr[i].index);
        }
    } )
    return answer;
}

File: 프로그래머스/lv3/42579. 베스트앨범/베스트앨범.py
def solution(genres, plays):
    answer = []

    dict_genre = {} # 장르별 총 재생수, {장르 : 총 재생 수}
    dict_plays = {} # 장르별 고유번호별 재생수, {장르 : [[고유번호:재생 수]]}

    for i in range(len(genres)) :
        if genres[i] in dict_genre :
            dict_genre[genres[i]] += plays[i]
            dict_plays[genres[i]].append([i,plays[i]])
        else :
            dict_genre[genres[i]] = plays[i]
            dict_plays[genres[i]] = [[i,plays[i]]]

    # 총 재생수 내림차순 정령
    sorted_dict = dict(sorted(dict_genre.items(), key=lambda item: -item[1]))

    # 개별 재생 수 내림차순 정렬 후 최대 2개 answer에 추가
    for key in sorted_dict :
        tmp = sorted(dict_plays[key],key=lambda a:-a[1])
        answer+=tmp[:2]

    return list(map(lambda a: a[0] ,answer))
